slasher/db/block_header.go

```diff
@@ -0,0 +1,114 @@
+package db
+
+import (
+	"bytes"
+
+	"github.com/boltdb/bolt"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+)
+
+func createBlockHeader(enc []byte) (*ethpb.BeaconBlockHeader, error) {
+	protoBlockHead := &ethpb.BeaconBlockHeader{}
+	err := proto.Unmarshal(enc, protoBlockHead)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to unmarshal encoding")
+	}
+	return protoBlockHead, nil
+}
+
+// BlockHeader accepts a block root and returns the corresponding block.
+// Returns nil if the block does not exist.
+func (db *Store) BlockHeader(epoch uint64, validatorID uint64) ([]*ethpb.BeaconBlockHeader, error) {
+	var bha []*ethpb.BeaconBlockHeader
+	err := db.view(func(tx *bolt.Tx) error {
+		c := tx.Bucket(historicBlockHeadersBucket).Cursor()
+		prefix := encodeEpochValidatorID(epoch, validatorID)
+		for k, v := c.Seek(prefix); k != nil && bytes.HasPrefix(k, prefix); k, v = c.Next() {
+			bh, err := createBlockHeader(v)
+			if err != nil {
+				return err
+			}
+			bha = append(bha, bh)
+		}
+		return nil
+	})
+	return bha, err
+}
+
+// HasBlockHeader accepts an epoch and validator id and returns true if the block header exists.
+func (db *Store) HasBlockHeader(epoch uint64, validatorID uint64) bool {
+	prefix := encodeEpochValidatorID(epoch, validatorID)
+	var hasBlockHeader bool
+	// #nosec G104
+	_ = db.view(func(tx *bolt.Tx) error {
+		c := tx.Bucket(historicBlockHeadersBucket).Cursor()
+		for k, _ := c.Seek(prefix); k != nil && bytes.HasPrefix(k, prefix); k, _ = c.Next() {
+			hasBlockHeader = true
+			return nil
+		}
+		hasBlockHeader = false
+		return nil
+	})
+
+	return hasBlockHeader
+}
+
+// SaveBlockHeader accepts a block header and writes it to disk.
+func (db *Store) SaveBlockHeader(epoch uint64, validatorID uint64, blockHeader *ethpb.BeaconBlockHeader) error {
+	key := encodeEpochValidatorIDSig(epoch, validatorID, blockHeader.Signature)
+	enc, err := proto.Marshal(blockHeader)
+	if err != nil {
+		return errors.Wrap(err, "failed to encode block")
+	}
+
+	err = db.update(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(historicBlockHeadersBucket)
+		if err := bucket.Put(key, enc); err != nil {
+			return errors.Wrap(err, "failed to include the block header in the historic block header bucket")
+		}
+
+		return err
+	})
+
+	// prune history to max size every 10th epoch
+	if epoch%10 == 0 {
+		weakSubjectivityPeriod := uint64(54000)
```

Comment:
can we put this in our config and then call it from there. Lets not hardcode this constant here

Reply:
moved, thanks